cargo-pgrx/src/command/install.rs

```diff
@@ -13,12 +13,19 @@ use crate::profile::CargoProfile;
 use crate::CommandExecute;
 use cargo_toml::Manifest;
 use eyre::{eyre, WrapErr};
+use once_cell::sync::Lazy;
 use owo_colors::OwoColorize;
 use pgrx_pg_config::{cargo::PgrxManifestExt, get_target_dir, PgConfig, Pgrx};
+use std::collections::HashMap;
 use std::fs;
 use std::io::BufReader;
 use std::path::{Path, PathBuf};
 use std::process::Stdio;
+use std::sync::{Arc, Mutex};
+
+/// Type used for memoizing expensive to calculate values.
+/// Arc<Mutex> is needed to get around compiler safety checks.
+type MemoizeKeyValue = Arc<Mutex<HashMap<String, String>>>;
```

Comment:
I don't think it matters in practice, and in this case it's more amenable to us later threadpooling the IO if it gets truly dreadful, but we could alternatively use https://doc.rust-lang.org/stable/std/macro.thread_local.html to defeat the borrow-checker's concerns.

Reply:
I'm down to try, I've never used thread_locals before.

Reply:
It's probably better to take a lock than to have a thread do IO we've already done.

Reply:
Yeah, I think this is fine here, I was only noting the alternate implementation choice because of my pedagogic compulsion.